src/centralserver/admin-service/src/main/java/org/niis/xroad/centralserver/restapi/service/CertificationServicesService.java

```diff
@@ -25,32 +25,58 @@
  */
 package org.niis.xroad.centralserver.restapi.service;
 
+import ee.ria.xroad.commonui.CertificateProfileInfoValidator;
+
 import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
 import org.niis.xroad.centralserver.openapi.model.ApprovedCertificationService;
 import org.niis.xroad.centralserver.restapi.converter.CertificationServiceConverter;
+import org.niis.xroad.centralserver.restapi.dto.ApprovedCertificationServiceDto;
 import org.niis.xroad.centralserver.restapi.entity.ApprovedCa;
 import org.niis.xroad.centralserver.restapi.repository.ApprovedCaRepository;
+import org.niis.xroad.restapi.config.audit.AuditDataHelper;
 import org.springframework.stereotype.Service;
 import org.springframework.transaction.annotation.Transactional;
 
 import java.util.List;
 import java.util.Set;
-import java.util.stream.Collectors;
 
+import static java.util.stream.Collectors.toSet;
+import static org.niis.xroad.restapi.config.audit.RestApiAuditProperty.AUTHENTICATION_ONLY;
+import static org.niis.xroad.restapi.config.audit.RestApiAuditProperty.CERTIFICATE_PROFILE_INFO;
+
+@Slf4j
 @Service
 @Transactional
 @RequiredArgsConstructor
 public class CertificationServicesService {
 
+    private final AuditDataHelper auditDataHelper;
     private final ApprovedCaRepository approvedCaRepository;
 
     private final CertificationServiceConverter certificationServiceConverter;
 
+    public ApprovedCertificationService add(ApprovedCertificationServiceDto approvedCa) {
```

Comment:
Well `add` method return class is not suffixed with `dto` as well because it is an openapi generated response resource class. The `dto` class is used to wrap attributes coming in to controller, purely used for convenience and data transfer. I'd say we still have the consistency, `dto` classes seem to be used in services as domain objects once remapped from request resource classes.